ValveResourceFormat/Resource/Resource.cs

```diff
@@ -232,59 +232,55 @@ public void Read(Stream input, bool verifyFileSize = true)
 
                 block.Offset = offset;
                 block.Size = size;
+                block.Resource = this;
 
                 Blocks.Add(block);
 
-                switch (block.Type)
+                if (block.Type is BlockType.RED2 or BlockType.REDI or BlockType.NTRO)
                 {
-                    case BlockType.REDI:
-                    case BlockType.RED2:
-                        block.Read(Reader, this);
+                    block.Read(Reader);
 
+                    if (block.Type is BlockType.RED2 or BlockType.REDI)
```

Comment:
feels like pointless nest, just make it two separate ones

Reply:
Done